### ISSUE TITLE
nanotidy and nanoimport 

### ISSUE THREAD

**Issue by Anara2018** [nguyens7/tidyNano#5]
Hello, 

I am having a problem while using nanotidy() function. My samples are uploaded with nano import (but each sample on which 5 trials were done on NTA) is uploaded separately and then i merged them. I am having a problem of seeing the dilution, technical replicate information, etc. when i upload these files. 

do you have a more detailed vignette or more explanations on what parameters could be used in these two functions with some examples? 
In the publication & the Readme file, the uploaded .csv file contains all the samples, whereas normally, typical NTA data comes in separate .csv file for each sample. 
thank you
best

**Comment by nguyens7** [comment]
I updated the `nanoimport()` function so it can extract the dilution information from the raw NTA .csv file and added this information in README.  If you didn't add the dilution factor in your sample name then you can put `auto_name == TRUE` in the `nanoimport()` function.  I added another beads2.csv dataset that doesn't have the dilution factor in the sample name (like in the beads.csv file).

See if you can reinstall tidyNano version 0.1.2.3 and run the following code.
> `# devtools::install_github("nguyens7/tidyNano")`
> library(tidyNano)
> library(tidyverse)

> file2 <- system.file("extdata", "beads2.csv", package = "tidyNano")

> data2 <- nanoimport(file2, auto_name = TRUE)

> data2 %>%
  nanotidy(sep_var = c("sample", "tech_rep", "Dilution")) %>% 
  ggplot(aes(particle_size, True_count, color = tech_rep)) +
  geom_line(size = 1, alpha = 0.8)`

Let me know if this addresses the issue!

**Comment by Anara2018** [comment]
Hello,
thank you for a quick response. 
So, i have re-installed tidyNano as you suggested. I am opening my .csv file as follows (my raw NTA .csv files have the dilution information from acquisition, so there's no need to include auto_name==TRUE). However, when I run nanotidy(), i am not able to get the dilution, true count, technical replicate information. 

Here is the code: 

> data285 <- nanoimport("285.csv")
NTA version: 3.1
Sample name: 
Dilution line no: 41
Dilution factor detected: 1000
Auto name = FALSE
Custom name: NULL
Dilution value: 1 (Didn't parse)


> data285 %>%
+   nanotidy(sep_var = c("sample", "tech_rep", "Dilution")) %>%
+   ggplot(aes(particle_size, True_count)) +
+   geom_line(size = 1, alpha = 0.8)
Warning messages:
1: Expected 3 pieces. Missing pieces filled with `NA` in 5000 rows [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, ...]. 
2: Removed 5000 row(s) containing missing values (geom_path). 
once imported, my .csv file only consists of 6 rows (1st: particle size; 2:6 for the 5 runs on the same sample). 

I am also attaching my .csv file, in case you can pinpoint inherent issue with the raw NTA data...? 

[285.csv.zip](https://github.com/nguyens7/tidyNano/files/4798542/285.csv.zip)

**Comment by nguyens7** [comment]
Hi, thanks for providing a reprex!  

I was able to take a look at the file and think I know what the issue was.  

Since the dilution factor was explicitly stated in the sample name (285 1-6 2020-01-23 11-37-07 **_1000** ) where the bolded section would represent the dilution factor we will have to obtain it and append it with the `auto_name = TRUE` argument. 

After we import with `nanoimport()` with `auto_name = TRUE` it will append `_1000` to all of your samples and we can use that to account for the dilution.   It seems like the issue from the `nanotidy()` function was the `sep_var` arguments.  Since we essentially have two pieces of information there (your sample information and the dilution factor) nanoimport will only parse out information that is separated by a `_` so we can just specify `c("sample_ID", "Dilution")` and that should allow you to move on to plotting.  

Here's the code that I used to get a plot, does this work for you?
`nanoimport("285.csv", auto_name = TRUE) %>%`
`  nanotidy(sep_var = c("sample_ID", "Dilution")) %>% `
 `ggplot(aes(particle_size, True_count, color = sample_ID)) +`
`  geom_line()`

![2020-06-18_tidyNano](https://user-images.githubusercontent.com/18275928/85028366-fbe3e680-b148-11ea-84c9-e8189687014e.png)
